### ISSUE TITLE
[BUG] Cannot connect when there are 2 consoles on the network (whether PS5 or PS4)

### ISSUE THREAD

**Issue by jvila-turnto** [ywnico/vitaki-fork#6]
**Describe the bug**
when 2 ps5 systems are in the same network and you select the first one the app just breaks! 

**To Reproduce**
open app
2 devices shown
select first device
app closes

**Expected behavior**
start working with the first ps5 and ask for code or start remote play

**Screenshots**

**Desktop (please complete the following information):**
 - OS: psvita
 - chiaki last version of this repo

**Additional context**

**Comment by ywnico** [comment]
Thanks for the report!

I'll try to investigate when I get the chance. Although unfortunately I don't have two PS5s so I'm not sure if I'll be able to debug this one....

I may need to develop some debug logging to file first....

**Comment by makerender** [comment]
Just wanted to confirm, two devices on the network breaks the app. @ywnico It happens with ps4s on the network as well so if you have one lying around it should be possible to reproduce.

**Comment by ywnico** [comment]
Thanks for the info! Unfortunately I don't have a ps4 either, but that provides some additional clues.

Just to note, I will probably not get around to more development within the next month or so since I'm kinda busy. Sorry for the delay. But more information like this is always helpful and appreciated.

**Comment by ywnico** [comment]
I've looked into it a bit, but I just don't see a good way to debug this without having 2 consoles myself. I hope someone with 2 consoles can debug it and submit a PR....

**Comment by xMrMurderx** [comment]
So I'm currently in this exact situation. I have my sons and my console on the network. The odd thing I noticed when I first opened your app was that when my ps5 was powered off, not in sleep mode, it would show my son's console twice in the list.

Anyway, sons console was in sleep mode, mine was powered on. I selected my console, vitaki froze for about 15 seconds then said "press x to pair console" and just hung after that. I let it sit for 10+ minutes with no update.

So I'm all in on helping. What exactly do you need?

*edit*

Fiddled around a little bit more tonight.  So the touchscreen is very finicky. I have to press about 10 times to select my ps5, then after it is selected nothing happens if I touch anything else, hence why I thought the app froze.

Using the dpad, I can select both consoles no problem.  The issue it seems is that the very first console in the list cannot be selected no matter what. The console gets highlighted, but hitting X does nothing.  If I move to my son's console (second one listed in the app) and hit X, it brings up the registration page.  Nothing in the logs to explain why this happens, it just shows it detects both ps5s and that's it.

**Comment by ywnico** [comment]
Thanks for all the details, @xMrMurderx, these are really good to know.

Unfortunately debugging is such an iterative process that I don't really have confidence in solving this unless I get a second console to test on. I'm really hoping someone else with 2 consoles (or someone with 1 console who's just smarter than me) will be able to clone the source code, solve the issue, and make a PR.

That said, it actually might be possible to work around this issue for now by turning off discovery and connecting directly by IP address. This would entail:
1. Turn off one console (or just disconnect it from the network temporarily) so that Vitaki will work on yours.
2. Go through the registration process (select your console, pair with it, ensure remote play works).
3. Add a new remote host in Vitaki (button with plus icon) using the internal IP of your console (like 192.168.1.123 or whatever. You could use your router to assign a static IP so that this never changes).
4. Turn off discovery in Vitaki (wifi icon) and make sure you can still connect to your console (it should appear as a remote host icon, which looks the same as a discovered console but with no blue light).
5. Now it _should_ be possible to connect even without discovery. So you can turn on the other console again and keep connecting to your registered one directly without using discovery.

I'm not sure the above workaround would be effective, but I think it should work....

**Comment by xMrMurderx** [comment]
Yeah I had the same thoughts on powering off the second console. I'll give it a try tomorrow, thank you

**Comment by xMrMurderx** [comment]
![Image](https://github.com/user-attachments/assets/24bc490b-c90d-4835-a49a-e3cd065b0afd)

That was a success, thank you again.

**Comment by exsilium** [comment]
Just adding here, that I have PS4 and PS5 on the network.
- Both of the consoles were in sleeping state
- I started Vitaki and discovery was successful for both of the consoles; PS5 was listed as the second console (idx 1).
- I started the PS5 locally and went to `settings->system->remote play->pairing view` to have the code shown
- I started pairing on the Vitaki screen, added the code and it connected successfully.

In short, altough I have 2 consoles on the same network and subnet, I did not encounter the freezing issue described here. I used the release vpk published in this repo.

**Comment by makerender** [comment]
I just deleted and reinstalled the latest version of Vitaki from this repo, and tried again.

Yes, can confirm with a ps5 and ps4 on the network pairing does work for me now. Seems fixed.

**Comment by Aster404** [comment]
I had the same issue and upon trying it again after updating it was fixed! :)
